contrib/check-for-api-changes.sh

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env bash
+#
+# Checks the public API of crates, exits with non-zero if there are currently
+# changes to the public API not already committed to in the various api.txt
+# files.
+
+set -e
+
+export RUSTDOCFLAGS='-A rustdoc::broken-intra-doc-links'
+REPO_DIR=$(git rev-parse --show-toplevel)
+API_DIR="$REPO_DIR/api"
+
+main() {
+    check_we_have_nightly
+
+    pushd "$REPO_DIR/bitcoin" > /dev/null
+    cargo public-api --no-default-features --features=no-std > "$API_DIR/bitcoin/api-no-std.txt"
+    cargo public-api > "$API_DIR/bitcoin/api-default-features.txt"
+    cargo public-api --all-features > "$API_DIR/bitcoin/api-all-features.txt"
+    popd > /dev/null
+
+    pushd "$REPO_DIR/hashes" > /dev/null
+    cargo public-api --no-default-features > "$API_DIR/hashes/api-no-features.txt"
+    cargo public-api > "$API_DIR/hashes/api-default-features.txt"
+    cargo public-api --no-default-features --features=alloc > "$API_DIR/hashes/api-alloc.txt"
+    cargo public-api --all-features > "$API_DIR/hashes/api-all-features.txt"
+    popd > /dev/null
+
+    if is_dirty ; then
+        echo "You have introduced changes to the public API, commit the changes to api/ currently in your working directory"
+        exit 1
+    else
+        echo "No changes to the current public API"
+    fi
+}
+
+# Returns true if the api directory has been modified.
+is_dirty() {
+    if git diff --quiet api || git diff --cached api; then
```

Comment:
In 13c70fa8:

I think you need to `pushd`/`popd` into the `$REPO_DIR` before calling `git diff`. Otherwise this script does not work from other directories.

Reply:
Good point, will fix. Thanks